### DISCUSSION TITLE
How to run a spoon / task on login?

### DISCUSSION THREAD

**Discussion by homerjam** [Hammerspoon/hammerspoon#3426] [Q&A] [answered]
Hi,

I'm trying to use hammerspoon on computer startup and I'm trying to find the best way to run tasks (as if a hotkey had been pressed) after login.

Is there any suggested way to do this?

Thanks!

**Comment by cmsj** [comment, marked as answer]
If you configure Hammerspoon to start at login, anything in your init.lua will be run. Is that sufficient? Or do you need a way to ensure that those things are only run right after a login, and not run again if Hammerspoon is restarted?
